### ISSUE TITLE
Cannot reproduce reported imagenet validation accuracy

### ISSUE THREAD

**Issue by is-cs** [wandering007/nasnet-pytorch#1]
<pre>
Hi
I seek help as I am unable to reproduce your reported results. Here is my setup:
Python version 3.6.5
Pytorch version 1.1.0

Imagenet validation dataset downloaded from imagenet website and placed in a val folder having the following structure:<br />
imagenet_dataset
    val/
        n01440764/
            ...images...
        n01443537/
            ...images...
        .... 1000 such folders

nasnet_a_large.pth weights downloaded from your dropbox link.

I run your script as follows:
python imagenet_eval.py --nas-type large --resume ../nasnet_a_large.pth --gpus 0 1 --data ../../data/imagenet_dataset --batch-size 4

I get Top1 = 17% and Top5 = 4%
Test set[50000]: Top1: 17.00%, Top5: 4.00%, Average loss: 0.8056

Please help.
Thank you
</pre>

**Comment by wandering007** [comment]
@is-cs Hi! The issue is caused by some code incompatibilities between 0.3 and 0.4(or high). `imagenet_eval.py` works for PyTorch 0.3 or lower. **I update my code in a rush just now.** You can check its correctness for me. Thanks :-). Besides, the migration guide is provided here: https://pytorch.org/blog/pytorch-0_4_0-migration-guide/

**Comment by is-cs** [comment]
I did use .item instead of .data[0]. But I see in your code that you are doing
top1 += len(data) - topk_correct.narrow(1, 0, 1).sum()
top5 += len(data) - topk_correct.sum()
This subtraction from len(data) i believe will calculate the number of incorrect samples. So 17% and 4% are essentially incorrect percentages. Percentage correct is top1=(100 - 17 = 83) and top5=(100 - 4 = 96), which very closely matches your reported results. Am i correct?

**Comment by wandering007** [comment]
@is-cs I see. The top1 and top-5 in the code refer to the error rates, not accuracies. Sorry for not clarifying it.
